omx/gstomxvideodec.c

```diff
@@ -2058,14 +1679,61 @@ gst_omx_video_dec_set_format (GstVideoDecoder * decoder,
   gst_omx_port_set_flushing (self->dec_in_port, 5 * GST_SECOND, FALSE);
   gst_omx_port_set_flushing (self->dec_out_port, 5 * GST_SECOND, FALSE);
 
+  if (gst_omx_port_populate (self->dec_out_port) != OMX_ErrorNone)
+    return FALSE;
+  if (gst_omx_port_mark_reconfigured (self->dec_out_port) != OMX_ErrorNone)
+    return FALSE;
+
   if (gst_omx_component_get_last_error (self->dec) != OMX_ErrorNone) {
     GST_ERROR_OBJECT (self, "Component in error state: %s (0x%08x)",
         gst_omx_component_get_last_error_string (self->dec),
         gst_omx_component_get_last_error (self->dec));
     return FALSE;
   }
 
+  /* Start the srcpad loop again */
+  GST_DEBUG_OBJECT (self, "Starting task again");
+
+  self->downstream_flow_ret = GST_FLOW_OK;
+  gst_pad_start_task (GST_VIDEO_DECODER_SRC_PAD (self),
+      (GstTaskFunction) gst_omx_video_dec_loop, decoder, NULL);
+
+  return TRUE;
+}
+
+static gboolean
+gst_omx_video_dec_reset (GstVideoDecoder * decoder, gboolean hard)
+{
+  GstOMXVideoDec *self;
+
+  self = GST_OMX_VIDEO_DEC (decoder);
+
+  /* FIXME: Handle different values of hard */
```

Comment:
The boolean hard is not even used, it seems that there's only one method to reset the component, so this boolean could be removed.

Reply:
Ready for review 0149b4f